src/transaction/transaction.rs

```diff
@@ -71,9 +79,38 @@ impl Transaction {
     /// ```
     pub async fn batch_get(
         &self,
-        _keys: impl IntoIterator<Item = impl Into<Key>>,
+        keys: impl IntoIterator<Item = impl Into<Key>>,
     ) -> Result<Vec<KvPair>> {
-        unimplemented!()
+        let mut result = Vec::new();
+        let mut keys_from_snapshot = Vec::new();
+        let mut result_indices_from_snapshot = Vec::new();
```

Comment:
Tt's not convenient to use a `Vec<(KvPair, usize)>` now. If we want to avoid cloning the key, we have to consume the `Vec` to get an iterator. Meanwhile, the indices are also consumed but we want to use them sometime later.